### ISSUE TITLE
Cannot delete gist file with filename: null

### ISSUE THREAD

**Issue by UziTech** [octokit/rest.js#1629]
according to the [documentation](https://developer.github.com/v3/gists/#edit-a-gist)
> To delete a file, set the value of the filename to `null`.

This gives the error `Invalid request. For 'properties/filename', nil is not a string.`

```js
octokit.gists.update({
  "gist_id": [REDACTED],
  "files": {
    "README": {
      "filename": null
    }
  }
})
```

**Comment by UziTech** [comment]
Settings content to an empty string does work to delete the file

```js
octokit.gists.update({
  "gist_id": [REDACTED],
  "files": {
    "README": {
      "content": ''
    }
  }
})
```

**Comment by gr2m** [comment]
Thanks for reporting the issue, I'll look into it

**Comment by gr2m** [comment]
Thanks Tony, I was able to reproduce the error:
https://runkit.com/gr2m/octokit-rest-js-1629

My guess is a problem with `@octokit/request`, I'll try to further narrow down the test case

**Comment by gr2m** [comment]
Yeah it looks like the problem is somewhere in `@octokit/request`:

https://runkit.com/gr2m/octokit-rest-js-1629/2.0.0

I've created another check to see if it's even lower in `@octokit/endpoint`, but that works

https://runkit.com/gr2m/octokit-rest-js-1629/3.0.0

I'll make a follow up issue in https://github.com/octokit/request.js/

**Comment by gr2m** [comment]
Turns out it's an actual API error

```
curl -X PATCH -H"Authorization: token [REDACTED]" https://api.github.com/gists/9f00c3cef38a7db78ee5c26da9f6db89 -d '{"files":{"test.md":{"content":null}}}'
{
  "message": "Invalid request.\n\nFor 'properties/content', nil is not a string.",
  "documentation_url": "https://developer.github.com/v3/gists/#edit-a-gist"
}
```

I'll let the right people know :)

**Comment by UziTech** [comment]
Has the API been fixed?

**Comment by ducktrshessami** [comment]
Just got the error using a different client today, so I'd say no.
> 422 - {\"message\":\"Invalid request.\\n\\nFor 'properties/filename', nil is not a string.\",\"documentation_url\":\"https://developer.github.com/v3/gists/#edit-a-gist\"}

**Comment by gr2m** [comment]
No, not yet. You can set the content to an empty string to delete a file for the time being

**Comment by gr2m** [comment]
The API was still not fixed, I'm afraid
